test/integration/singlecluster/scheduler/suite_test.go

```diff
@@ -100,3 +116,44 @@ func managerAndSchedulerSetup(ctx context.Context, mgr manager.Manager) {
 	err = sched.Start(ctx)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 }
+
+func setupInterceptedClient() (context.Context, client.Client) {
```

Comment:
btw, I think it is nice to see this PR paves the way for simulating API server errors. We have some older code which is not tested wrt error handling because we never added the mechanism. For example https://github.com/kubernetes-sigs/kueue/pull/7364, but many more in the past.

Now I can refer to this PR as an example for covering error cases in integration tests. Thanks ! 

cc @PBundyra @pajakd .